### DISCUSSION TITLE
ESP32 + Contact Door Sensor + Relay

### DISCUSSION THREAD

**Discussion by hhorigian** [arendst/Tasmota#20084] [Q&A] [answered]
Hi Everyone - 
I know there are many discussions on this matter, but could not find a simple solution to this simple problem.
I am having ESP32 Dev kits, with a simple door sensor (GPIO13+GND) , a DH1820 (GPIO15) and a 2 Channel Relay Module (GPIO19 = Relay).  Cannot get more simple than that.
Installed latest sensors firmware. Idea is to send MQTT report status and to receive relay commands. 

I know that the "Switch"in tasmota are attached to the Relays, so the minute I setup the GPIO13 as a Switch, and activate/deactivate that door sensor contact, the Relay will turn on/off. 
(Tried Rules, Tried switchmode, and some other examples on the forum, but something that should be so simple - I could nail it). 

So what would be the solution or config to having the "door sensor"/contact  working together with the Relay and not conflicting? 
Appreciate any help.
Thanks!

**Comment by barbudor** [comment, marked as answer]
Rules, setoption114 or switchmode 15 all work fine
As you didn't provide details on what you tried exactly so hard to tell what you missed

As you don't tell neither what you want to happens when the door switch change states, hard to tell more

**Reply by hhorigian** [reply]
Hi @barbudor - thanks for the reply.

Didn't know about the setoption114.  That did the trick. 
setoption114 on 

Before that, I tried rules, and tried the switchmode 15 too but didn't help. 

For reference. I didn't want anything to happen when the door switch change. Just report to MQTT the status change, but no rules or anything. Just status.
Thanks again.

**Reply by barbudor** [reply]
Hi @hhorigian 
Switchmode<x> 15 should also do something similar but with a slightly different MQTT reporting
You may have not used the proper syntax

**Reply by sfromis** [reply]
Likewise, it *can* also work with rules doing `Publish`, **if** you use a suitable combination of `SwitchMode` and values, but in general, `SetOption114` is indeed the simplest if you just want MQTT messages for all Switch inputs, where `SwitchMode 15` is per channel.
